server/core/src/server_utilities.cpp

```diff
@@ -39,5 +39,24 @@ namespace irods
 
         return std::regex_search(_rule_text.data(), session_vars_pattern);
     }
+
+    auto to_bytes_buffer(const std::string_view _s) -> BytesBuf*
+    {
+        constexpr auto allocate = [](const auto bytes) noexcept
+        {
+            return std::memset(std::malloc(bytes), 0, bytes);
+        };
+
+        const auto buf_size = _s.length() + 1;
+
+        auto* buf = static_cast<char*>(allocate(sizeof(char) * buf_size));
+        std::snprintf(buf, buf_size, "%s", _s.data());
```

Comment:
What is the reasoning behind using `snprintf` over `strncpy` here?

Reply:
Codacy complained via the FlawFinder tool which states: "Easily used incorrectly; doesn't always \0-terminate or check for invalid pointers [MS-banned] (CWE-120)." `std::snprintf` seems to be considered a safe alternative.

Reply:
That makes sense. However, I bet this likely incurs more overhead. Maybe codacy would feel better about `std::memcpy`. In this case, we don't have to pay for the additional overhead given `buf` is cleared to zeros which gives us the null terminator for free.

Reply:
Assuming my testing is correct, I'm showing strncpy as being 4.4x faster than snprintf and memcpy being 4.3x faster than snprintf. Here is the test: https://quick-bench.com/q/axHaBt3rn3B5FIX_h5QSexwCnBM

Reply:
The only problem I see with `strncpy` is that it stops at null bytes, but given that `to_bytes_buffer` takes in a `string_view`, it is probably safe to say that is fine.

Reply:
Cool, I don't mind using either one, really. I will try `std::memcpy` and see how that goes.

Reply:
Codacy has deemed `std::memcpy` acceptable. I am okay to revert this to use `std::strncpy` as well. If we go that route, I may leave a comment in defense of its use for future readers of the Codacy issues.

Reply:
Cool. It's your call.

Reply:
Okay. I think I'll just revert it and leave an explanatory comment. Gracias